src/implements.jl

```diff
@@ -67,3 +68,23 @@ end
 
 _all_in(items::Tuple, collection) = all(map(in(collection), items))
 _all_in(item::Symbol, collection) = in(item, collection)
+
+struct Implemented{T<:Interface} end
+struct NotImplemented{T<:Interface} end
+
+"""
+    implemented_trait(T::Type{<:Interface}, obj)
+    implemented_trait(T::Type{<:Interface{Option}}, obj)
+
+Provides a single type for using interface implementation
+as a trait.
+
+Returns `Implemented{T}()` or `NotImplemented{T}`.
+"""
+function implemented_trait(::Type{T}, obj) where T<:Interface
```

Comment:
I think the name is fine. Do we want to export this function and the related types?

Reply:
Currently only the macros are exported, Im not sure what the general approach should be

Reply:
Let's keep it inside for now, it's not crucial anyway and users will import that particular name if they need it.